src/FolderContext.ts

```diff
@@ -53,12 +53,12 @@ export class FolderContext implements vscode.Disposable {
     }
 
     private setContextKeys() {
-        if (this.swiftPackage.contents === undefined) {
-            contextKeys.hasPackage = false;
-            contextKeys.packageHasDependencies = false;
+        if (this.swiftPackage.foundPackage) {
```

Comment:
Shouldn't we use `isValid` here? Setting `hasPackage` will trigger the task provider and the dependency view, but neither will work properly if the package is invalid.

Reply:
I'm not sure either way. There was a point when `contextKeys.hasPackage` was the gate for running a resolve. Currently we need a resolve to run to report back errors in the `Package.swift`. We need to improve the error reporting there but that is for another time. Anyway changed to `isValid`